src/Backend/Modules/ContentBlocks/Actions/Add.php

```diff
@@ -10,91 +10,72 @@
         parent::execute();
-        $this->templates = BackendContentBlocksModel::getTemplates();
-        $this->loadForm();
-        $this->validateForm();
-        $this->parse();
-        $this->display();
-    }
 
-    /**
-     * Load the form
-     */
-    private function loadForm()
-    {
-        $this->frm = new BackendForm('add');
-        $this->frm->addText('title', null, null, 'form-control title', 'form-control danger title');
-        $this->frm->addEditor('text');
-        $this->frm->addCheckbox('hidden', true);
+        $createContentBlockCommand = new CreateContentBlock();
+        $form = $this->createForm(
+            new ContentBlockType(
+                CreateContentBlock::class,
+                $this->get('fork.settings')->get('Core', 'theme', 'core')
+            ),
+            $createContentBlockCommand
+        );
 
-        // if we have multiple templates, add a dropdown to select them
-        if (count($this->templates) > 1) {
-            $this->frm->addDropdown('template', array_combine($this->templates, $this->templates));
-        }
-    }
+        $form->handleRequest($this->get('request'));
 
-    /**
-     * Validate the form
-     */
-    private function validateForm()
-    {
-        if ($this->frm->isSubmitted()) {
-            $this->frm->cleanupFields();
-            $fields = $this->frm->getFields();
+        if (!$form->isValid()) {
+            $this->tpl->assign('form', $form->createView());
 
-            // validate fields
-            $fields['title']->isFilled(BL::err('TitleIsRequired'));
-            $fields['text']->isFilled(BL::err('FieldIsRequired'));
+            $this->parse();
+            $this->display();
 
-            if ($this->frm->isCorrect()) {
-                // build item
-                $item['id'] = BackendContentBlocksModel::getMaximumId() + 1;
-                $item['user_id'] = BackendAuthentication::getUser()->getUserId();
-                $item['template'] = count($this->templates) > 1 ? $fields['template']->getValue() : $this->templates[0];
-                $item['language'] = BL::getWorkingLanguage();
-                $item['title'] = $fields['title']->getValue();
-                $item['text'] = $fields['text']->getValue();
-                $item['hidden'] = $fields['hidden']->getValue() ? 'N' : 'Y';
-                $item['status'] = 'active';
-                $item['created_on'] = BackendModel::getUTCDate();
-                $item['edited_on'] = BackendModel::getUTCDate();
+            return;
+        }
 
-                // insert the item
-                $item['revision_id'] = BackendContentBlocksModel::insert($item);
+        $createContentBlockCommand = $form->getData();
 
-                // trigger event
-                BackendModel::triggerEvent($this->getModule(), 'after_add', array('item' => $item));
+        /** @var ContentBlock $contentBlock */
+        $contentBlock = $this->get('content_blocks.handler')->create($createContentBlockCommand);
```

Comment:
I think it's a convention to use `handle` as the method name of a command handler.